File: app/src/main/java/com/sumian/app/account/contract/LoginContract.java
package com.sumian.app.account.contract;

import com.sumian.app.base.BaseNetView;
import com.sumian.app.base.BasePresenter;
import com.sumian.app.network.request.LoginBody;

/**
 * Created by jzz
 * on 2017/9/26
 * <p>
 * desc:
 */

public interface LoginContract {

    interface View extends BaseNetView<Presenter> {

        void loginSuccess();

        void loginFailed(String error);

    }


    interface Presenter extends BasePresenter {

        void doLogin(LoginBody loginBody);
    }
}

File: doctor/src/main/java/com/sumian/sddoctor/login/login/LoginContract.kt
package com.sumian.sddoctor.login.login

import android.app.Activity
import android.content.Context
import com.sumian.sddoctor.base.BasePresenter
import com.sumian.sddoctor.base.BaseView

/**
 * <pre>
 *     @author : Zhan Xuzhao
 *     e-mail : xuzhao.z@sumian.com
 *     time   : 2018/6/14 20:01
 *     desc   :
 *     version: 1.0
 * </pre>
 */
class LoginContract {

    interface View : BaseView {
        fun launchMain()
        fun onRequestCaptchaSuccess()
        fun getContext(): Context
        fun onRequestCaptchaFail(code: Int)
    }

    interface Presenter : BasePresenter {
        fun loginByPassword(mobile: String, password: String)
        fun loginByCaptcha(mobile: String, captcha: String)
        fun loginByWechat(activity: Activity)
        fun requestCaptcha(mobile: String)
        fun requestCaptcha(mobile: String, captchaId: String, captchaPhrase: String)
    }
}